php/libraries/DicomTar.class.inc

```diff
@@ -0,0 +1,485 @@
+     *
+     * @return DicomTarDTO
+     */
+    public function withTarchiveid($tarchiveid): DicomTarDTO
+    {
+        $new = clone($this);
+        $new->_tarchiveid = $tarchiveid;
+        return $new;
+    }
+
+    /**
+     * Accessor for tarname
+     *
+     * @return string|null
+     */
+    public function getTarname(): ?string
+    {
+        return $this->_tarname;
+    }
+
+    /**
+     * Returns a new instance of this class with the new value
+     *
+     * @param string $tarname The new tarname
+     *
+     * @return DicomTarDTO
+     */
+    public function withTarname($tarname): DicomTarDTO
+    {
+        $new           = clone($this);
+        $new->_tarname = $tarname;
+        return $new;
+    }
+
+    /**
+     * Accessor for ArchiveLocation
+     *
+     * @return string|null
+     */
+    public function getArchiveLocation(): ?string
+    {
+        return $this->_archivelocation;
+    }
+
+    /**
+     * Returns a new instance of this class with the new value
+     *
+     * @param string $archivelocation The new ArchiveLocation
+     *
+     * @return DicomTarDTO
+     */
+    public function withArchiveLocation($archivelocation): DicomTarDTO
+    {
+        $new = clone($this);
+        $new->_archivelocation = $archivelocation;
+        return $new;
+    }
+
+    /**
+     * Accessor for patientname
+     *
+     * @return string|null
+     */
+    public function getPatientname(): ?string
+    {
+        return $this->_patientname;
+    }
+
+    /**
+     * Returns a new instance of this class with the new value
+     *
+     * @param string $patientname The new patientname
+     *
+     * @return DicomTarDTO
+     */
+    public function withPatientname(string $patientname): DicomTarDTO
+    {
+        $new = clone($this);
+        $new->_patientname = $patientname;
+        return $new;
+    }
+
+    /**
+     * Accessor for series
+     *
+     * @return DicomSerieDTO[]
+     */
+    public function getSeries(): array
+    {
+        return $this->_series;
+    }
+
+    /**
+     * Implements \LORIS\Data\DataInstance interface
+     *
+     * @return string
+     */
+    public function toJSON() : string
+    {
+        $series = array_map(
+            function ($item) {
+                return $item->toArray();
+            },
+            $this->_series
+        );
+
+        return json_encode(
+            array(
+             'tarchiveid'  => $this->_tarchiveid,
+             'tarname'     => $this->_tarname,
+             'patientname' => $this->_patientname,
+             'series'      => $series,
+            )
+        );
+    }
+}
+/**
+ * This class defines a DicomTarSerieDTO which is an immutable representation of a
+ * Dicom series object.
+ *
+ *  @category Imaging
+ *  @package  Main
+ *  @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ *  @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ *  @link     https://www.github.com/aces/Loris/
+ */
+class DicomSerieDTO
```

Comment:
again with `seriesdescription` and `seriesnumber`. Seems clearer to use these terms and does not take much more place in the code